### ISSUE TITLE
force:source:deploy with --json flag not showing failing test results

### ISSUE THREAD

**Issue by djs5008** [forcedotcom/cli#406]
### Summary

When attempting to deploy a failing test and retrieve the results via JSON, no test results are shown, and a stack trace pointing to ALMError is provided.

Command Used:
```
sfdx force:source:deploy --checkonly --testlevel RunSpecifiedTests --runtests ExampleTest --json -u example-alias -p force-app/main/default/classes/ExampleTest.cls
```

### Expected result

```
{
  "status": 1,
  "result": {
    "checkOnly": true,
    "completedDate": "2020-05-14T03:58:04.000Z",
    "createdBy": "0051N000006Olh5",
    "createdByName": "Example Name",
    "createdDate": "2020-05-14T03:57:03.000Z",
    "details": {
      "componentSuccesses": [
        {
          "changed": "false",
          "componentType": "ApexClass",
          "created": "false",
          "createdDate": "2020-05-14T03:57:30.000Z",
          "deleted": "false",
          "fileName": "sdx_sourceDeploy_1589428621777/classes/ExampleTest.cls",
          "fullName": "ExampleTest",
          "id": "01p3l00000EEpZLAA1",
          "success": "true"
        },
        {
          "changed": "true",
          "componentType": "",
          "created": "false",
          "createdDate": "2020-05-14T03:57:30.000Z",
          "deleted": "false",
          "fileName": "sdx_sourceDeploy_1589428621777/package.xml",
          "fullName": "package.xml",
          "success": "true"
        }
      ],
      "runTestResult": {
        "failures": [
          {
            "id": "01p3l00000EEpZLAA1",
            "message": "System.AssertException: Assertion Failed",
            "methodName": "testFail",
            "name": "Example Test",
            "namespace": {
              "$": {
                "xsi:nil": "true"
              }
            },
            "packageName": "ExampleTest",
            "stackTrace": "Class.ExampleTest.testFail: line 6, column 1",
            "time": "136.0",
            "type": "Class"
          }
        ],
        "flowCoverage": [
           ...
        ],
        "numFailures": "1",
        "numTestsRun": "5",
        "successes": [
           ...
        ],
        "totalTime": "14829.0"
      }
    },
    "done": true,
    "id": "0Af2g0000077QZdCAM",
    "ignoreWarnings": false,
    "lastModifiedDate": "2020-05-14T03:58:04.000Z",
    "numberComponentErrors": 0,
    "numberComponentsDeployed": 1,
    "numberComponentsTotal": 1,
    "numberTestErrors": 1,
    "numberTestsCompleted": 4,
    "numberTestsTotal": 5,
    "rollbackOnError": true,
    "runTestsEnabled": true,
    "startDate": "2020-05-14T03:57:28.000Z",
    "status": "Failed",
    "success": false
  }
}
```

### Actual result

```
{
  "status": 1,
  "name": "testFailure",
  "message": "The metadata deploy operation failed.",
  "exitCode": 1,
  "commandName": "SourceDeployCommand",
  "stack": "mdapiDeployFailed: The metadata deploy operation failed.\n    at ALMError (/Users/username/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/core/almError.js:50:19)\n    at MdDeployReportApi._throwErrorIfDeployFailed (/Users/username/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/mdapi/mdapiDeployReportApi.js:330:25)\n    at BBPromise.resolve.then.then.result (/Users/username/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/mdapi/mdapiDeployReportApi.js:286:34)\n    at tryCatcher (/Users/username/.local/share/sfdx/node_modules/bluebird/js/release/util.js:16:23)\n    at Promise._settlePromiseFromHandler (/Users/username/.local/share/sfdx/node_modules/bluebird/js/release/promise.js:517:31)\n    at Promise._settlePromise (/Users/username/.local/share/sfdx/node_modules/bluebird/js/release/promise.js:574:18)\n    at Promise._settlePromise0 (/Users/username/.local/share/sfdx/node_modules/bluebird/js/release/promise.js:619:10)\n    at Promise._settlePromises (/Users/username/.local/share/sfdx/node_modules/bluebird/js/release/promise.js:699:18)\n    at _drainQueueStep (/Users/username/.local/share/sfdx/node_modules/bluebird/js/release/async.js:138:12)\n    at _drainQueue (/Users/username/.local/share/sfdx/node_modules/bluebird/js/release/async.js:131:9)\n    at Async._drainQueues (/Users/username/.local/share/sfdx/node_modules/bluebird/js/release/async.js:147:5)\n    at Immediate.Async.drainQueues [as _onImmediate] (/Users/username/.local/share/sfdx/node_modules/bluebird/js/release/async.js:17:14)\n    at runCallback (timers.js:705:18)\n    at tryOnImmediate (timers.js:676:5)\n    at processImmediate (timers.js:658:5)\nOuter stack:\n    at Function.wrap (/Users/username/.local/share/sfdx/node_modules/salesforce-alm/node_modules/@salesforce/core/lib/sfdxError.js:151:27)\n    at MdDeployApi._doDeploy (/Users/username/.local/share/sfdx/node_modules/salesforce-alm/dist/lib/mdapi/mdapiDeployApi.js:252:40)",
  "warnings": []
}
```

### Additional information

When manually downgrading the `salesforce-alm` plugin to version 48.10.0 I receive the correct JSON response (the response I posted under Expected Result).

This error only shows with tests that fail. If all tests pass, I also receive the correct JSON response.

**SFDX CLI Version**:
`sfdx-cli/7.57.1-dbf54fc584 darwin-x64 node-v10.15.3`

**SFDX plugin Version**:
![image](https://user-images.githubusercontent.com/35700921/81892204-7fbe1800-9578-11ea-801d-b8efece3825b.png)

**OS and version**:
macOS Catalina v10.15.4

**Comment by VenkatKoppolu** [comment]
Even my team is facing this similar issue the stack trace looks ok but then the exception leads to nowhere with the script and we are unable to capture the exact error due to this, we had to put some logging to verify the JSON but seems weird.

**Comment by mshanemc** [comment]
fixed 7/8/2020 as W-7620800
